addons/mail/static/tests/discuss_app/discuss_tests.js

```diff
@@ -2076,3 +2082,74 @@ QUnit.test(
+        const [partnerId1, partnerId2] = pyEnv["res.partner"].create([
+            { name: "Mario" },
+            { name: "Mama" },
+        ]);
+        pyEnv["res.users"].create([{ partner_id: partnerId1 }, { partner_id: partnerId2 }]);
+        const deferred1 = makeDeferred();
+        const deferred2 = makeDeferred();
+        const { openDiscuss } = await start({
+            async mockRPC(route, params) {
+                if (route === "/web/dataset/call_kw/res.partner/im_search") {
+                    const { args } = params;
+                    if (args[0] === "m") {
+                        assert.step("First RPC");
+                        await deferred1;
+                    }
+                    if (args[0] === "mar") {
+                        assert.step("Second RPC");
+                        await deferred2;
+                    }
+                }
+            },
+        });
+        await openDiscuss();
+        await click(".o-mail-DiscussSidebar i[title='Start a conversation']");
+        await insertText(".o-mail-ChannelSelector input", "m");
+        await waitUntil(".o-mail-NavigableList-item:contains(Loading)");
+        await insertText(".o-mail-ChannelSelector input", "a");
+        await insertText(".o-mail-ChannelSelector input", "r");
+        deferred1.resolve();
+        assert.verifySteps(["First RPC"]);
+        await waitUntil(".o-mail-ChannelSelector-suggestion:contains(Mario)");
+        await waitUntil(".o-mail-ChannelSelector-suggestion:contains(Mama)");
+        deferred2.resolve();
+        assert.verifySteps(["Second RPC"]);
+        await waitUntil(".o-mail-ChannelSelector-suggestion:contains(Mama)", 0);
+        await waitUntil(".o-mail-ChannelSelector-suggestion:contains(Mario)");
+    }
+);
+
+QUnit.test(
+    "Channel input should wait until the previous RPC is done before starting a new one",
+    async (assert) => {
+        const pyEnv = await startServer();
+        pyEnv["discuss.channel"].create({ name: "Mario" });
+        const deferred = makeDeferred();
+        const { openDiscuss } = await start({
+            async mockRPC(route, params) {
+                if (route === "/web/dataset/call_kw/discuss.channel/search_read") {
+                    const { kwargs } = params;
+                    if (kwargs.domain[1][2] === "m") {
+                        assert.step("First RPC");
+                        await deferred;
+                    }
+                }
+            },
+        });
+        await openDiscuss();
+        await click(".o-mail-DiscussSidebar i[title='Add or join a channel']");
+        await insertText(".o-mail-ChannelSelector input", "m");
+        await insertText(".o-mail-ChannelSelector input", "a");
+        await insertText(".o-mail-ChannelSelector input", "r");
+        deferred.resolve();
+        await waitUntil(".o-mail-ChannelSelector-suggestion:contains(Mario)");
+        assert.verifySteps(["First RPC"]);
+    }
```

Comment:
I would remove this test. This is basically duplicate of previous test, except it uses channel instead of chat.
If we have some code shared among dozens of inputs, it's not worth making dozens of very similar tests (1 test gives a lot of gain, all others tests have barely any gain but adds significant maintenance cost in tests)